share/eds/byzantine/share_proof.go

```diff
@@ -24,31 +27,31 @@ type ShareWithProof struct {
 	share.Share
 	// Proof is a Merkle Proof of current share
 	Proof *nmt.Proof
+	// Axis is a proof axis
+	Axis rsmt2d.Axis
 }
 
-// NewShareWithProof takes the given leaf and its path, starting from the tree root,
-// and computes the nmt.Proof for it.
-func NewShareWithProof(index int, share share.Share, pathToLeaf []cid.Cid) *ShareWithProof {
-	rangeProofs := make([][]byte, 0, len(pathToLeaf))
-	for i := len(pathToLeaf) - 1; i >= 0; i-- {
-		node := ipld.NamespacedSha256FromCID(pathToLeaf[i])
-		rangeProofs = append(rangeProofs, node)
+// Validate validates inclusion of the share under the given root CID.
+func (s *ShareWithProof) Validate(dah *share.Root, axisType rsmt2d.Axis, axisIdx, shrIdx int) bool {
+	var rootHash []byte
+	switch axisType {
+	case rsmt2d.Row:
+		rootHash = rootHashForCoordinates(dah, s.Axis, shrIdx, axisIdx)
+	case rsmt2d.Col:
+		rootHash = rootHashForCoordinates(dah, s.Axis, axisIdx, shrIdx)
```

Comment:
I'm evaluating your changes to port them into Rust's `celestia-types`.

Correct me if I'm wrong but isn't this the same as the following?

```
	switch s.Axis {
	case rsmt2d.Row:
		rootHash = dah.RowRoots[axisIdx]
	case rsmt2d.Col:
		rootHash = dah.ColumnRoots[axisIdx]
```